Charades_Game_In_JavaScript/Charades Game In JavaScript final blog.md

```diff
@@ -0,0 +1,348 @@
+Besides the design of cards, there are some important features to add to make the card flips as we want. I mainly adapted these [CSS codes](https://github.com/code-sketch/memory-game/blob/master/video-11/styles.css) from [Marina](https://medium.com/free-code-camp/vanilla-javascript-tutorial-build-a-memory-game-in-30-minutes-e542c4447eae), because she has made the flipping effects very fluent and nice. However, instead of flipping it vertically, I changed to horizontally, so we have `transform: rotateX(180deg);`
+
+You can find my full CSS code [here](https://github.com/bitprj/Charades_Game_In_JavaScript/blob/master/finish_code/cardstyle.css).
+
+![](https://i.imgur.com/r8J1YZg.gif)
+
+
+Note: Here is a little tricky. Since I used the same CSS codes from Marina, the `back-face` is shown in the beginning as the first word, so I named it with `id="word1"`. I'll refer to `word1`, `word2` instead to make it more clear.
+
+
+### "Limiting Time" Ver.
+In this version, we want to have a timer to keep track of the time that each card has been shown, and flip the card when the time is up.
+To do this, we would write a function `function flipWord() {...}` which flips the word, and then start the timer. When the time is up, call `flipWord()` again and reset the `textContent` of the word whichever is hidden now, and it would be the next word to be shown.
+
+##### **in JavaScript:**
+**1. the First Flip**
+The first flip is different from the rest, because `word2` is already there and we do not need to reset any word. So we could simply use the same `setInterval()` function in our previous timer and call `flipWord()` instead of jumping to the next window. Set the seconds to 30 instead.
+
+*check out the timer implementation in phase 2 in the "Challenge: Adding a Countdown Before Redirecting" section under "Choosing the Game’s Rule" if you skipped that part*
+
+![](https://i.imgur.com/9eTYQo6.png)
+
+
+**2. `flipCard()` Function**
+When we call `flipCard()`, we want to to set the style corresponding to the "flip" we declared in the CSS file. This is similar to the "Hide and Show" strategy we used in the previous section. *Go back and check that phase 1 out!* 
+So we would simply have one line here. `document.getElementById("words").classList.toggle('flip');` 
+
+**3. Keep Flipping Cards and Resetting Words**
+After the first flip, a new 30 seconds countdown should start, so we add the same timer again in the `flipWord()` function after `toggle('flip);'`. And now we are seeing `word2` on screen, and `word1` is being hidden in the back. This is the time to change `word1` secretly! How would we do that?
+
+While it is definitely an idea to creating a number of the same card's HTMLs and link them one by one to jump to the new words, here we will keep the single HTML file and add a counter to keep track of the new word.
```

Comment:
do you think it's necessary to include the idea you aren't using? You could maybe delete the part before "here we will.." since the info before isn't used here. Up to you!